docs/yaml-classes-and-relationships.rst

```diff
@@ -702,6 +708,12 @@ editable
   :Type: boolean
   :Default Value: false
   
+optional
+  :Description: If component is user_edited, and editable is true, is this an optional attribute?
```

Comment:
If I understand what you're saying, you wanted this "optional" boolean to control whether a field shows up on the add dialog or not. It has nothing to do with editing on the details panel?

If so, I'd name it "add_dialog_display" to continue the theme of "grid_display" and "details_display".